elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreTransaction.java

```diff
@@ -61,27 +70,41 @@ public void createObject(Object entity, RequestScope scope) {
 
     @Override
     public Iterable<Object> loadObjects(EntityProjection entityProjection, RequestScope scope) {
-        Query query = buildQuery(entityProjection, scope);
         QueryResult result = null;
-
+        QueryResponse response = null;
         String cacheKey = null;
-        if (cache != null && !query.isBypassingCache()) {
-            String tableVersion = queryEngine.getTableVersion(query.getTable(), queryEngineTransaction);
-            if (tableVersion != null) {
-                cacheKey = tableVersion + ';' + QueryKeyExtractor.extractKey(query);
-                result = cache.get(cacheKey);
+        try {
+            aggregationAcceptQuery(scope);
+            Query query = buildQuery(entityProjection, scope);
+            if (cache != null && !query.isBypassingCache()) {
+                String tableVersion = queryEngine.getTableVersion(query.getTable(), queryEngineTransaction);
+                if (tableVersion != null) {
+                    cacheKey = tableVersion + ';' + QueryKeyExtractor.extractKey(query);
+                    result = cache.get(cacheKey);
+                }
             }
-        }
-        if (result == null) {
-            result = queryEngine.executeQuery(query, queryEngineTransaction);
-            if (cacheKey != null) {
-                cache.put(cacheKey, result);
+            aggregationProcessQuery(result, entityProjection.getType().getName(),
+                    query, scope);
+            if (result == null) {
+                result = queryEngine.executeQuery(query, queryEngineTransaction);
+                if (cacheKey != null) {
+                    cache.put(cacheKey, result);
+                }
             }
+            if (entityProjection.getPagination() != null && entityProjection.getPagination().returnPageTotals()) {
+                entityProjection.getPagination().setPageTotals(result.getPageTotals());
+            }
+            response = new QueryResponse(200, result.getData(), null);
```

Comment:
There are enums with the HTTP status codes (Like HttpStatus).

Reply:
Resolved in latest commit